util/coding.h

```diff
@@ -254,6 +261,15 @@ inline void PutVarint64(std::string* dst, uint64_t v) {
   dst->append(buf, static_cast<size_t>(ptr - buf));
 }
 
+inline void PutVarsignedint64(std::string* dst, int64_t v) {
+  char buf[10];
+  // Shift the absolute number right for one bit. Then use the least significant
+  // bit to indicate the sign.
+  char* ptr =
+      v >= 0 ? EncodeVarint64(buf, v * 2) : EncodeVarint64(buf, v * -1 * 2 + 1);
```

Comment:
Wouldn't `EncodeVarint64(buf, v >= 0 ? uint64_t(v) * 2 : -uint64_t(v) * 2 + 1 )` be simpler?

Also be careful with the casts and their order, you're doing signed overflow here (note that `-INT_MIN` overflows, so you have to negate after switching to unsigned)

Also, have you considered using the branch-free routines used in Thrift? They're probably much faster if the sign is unpredictable:

https://github.com/facebook/fbthrift/blob/449a5f77f9f9bae72c9eb5e78093247eef185c04/thrift/lib/cpp/util/VarintUtils-inl.h#L202-L208